app/assets/javascripts/modules/remove-filter.js

```diff
@@ -57,10 +56,9 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         var haystack = ' ' + currentVal.trim() + ' '
         var needle = ' ' + decodeEntities(removeFilterValue.toString()) + ' '
         var newVal = haystack.replace(needle, ' ').replace(/ {2}/g, ' ').trim()
-
-        $input.val(newVal).trigger(onChangeSuppressAnalytics)
+        window.GOVUK.triggerEvent($input.val(newVal)[0], 'change', suppressAnalytics)
```

Comment:
The suppressAnalytics options will need to be passed in as a detail argument:

```
window.GOVUK.triggerEvent($input[0], 'change', { detail: { suppressAnalytics: true } )
```

Otherwise that argument won't be sent to IE (https://github.com/alphagov/govuk_publishing_components/blob/98aea354ba78e9b242b097fefd97ce015ffe9a69/app/assets/javascripts/govuk_publishing_components/lib/trigger-event.js#L15) I also don't think it works on modern browsers either:

```
e = new CustomEvent('change', { detail: { detailOption: true }, nonDetailOption: true })
=> CustomEvent {isTrusted: false, detail: {…}, type: "change", target: null, currentTarget: null, …}
e.detail.detailOption
=> true
e.nonDetailOption
=> undefined
```

Once that change is made we'll need to adapt some other code which receives the event so it can handle a non jQuery event. There's code in this commit that you can use for that handling: https://github.com/alphagov/finder-frontend/commit/efb52d2086fab43db08fc1aaf59aa2556e38d9c6

@andysellick should the parameters option on https://github.com/alphagov/govuk_publishing_components/blob/98aea354ba78e9b242b097fefd97ce015ffe9a69/app/assets/javascripts/govuk_publishing_components/lib/trigger-event.js#L5 actually be just a detail option? as that's the only key you could set on the argument that will actually affect IE - any other properties set will be silently dropped.

Reply:
That's true, IE won't receive any other parameters. But we're starting to move away from IE a little, in that more things are going to be a non-JS fallback as the browser support drops. Since other browsers can understand other parameters, I think we should keep that script as it is for now, but we may need to raise awareness among the community of this limitation.

Reply:
Gotcha, I'd have thought that if you wanted to do something IE didn't support then you'd probably use CustomEvent directly and you'd only use that function if you wanted it to be IE compatible. It could be a bit surprising for other CustomEvent usages since it sets bubbles and cancelable which you might not expect.

Reply:
Yeah, the script could do with a bit of work. It's only being used in a few places right now, but something I need to monitor and expand on.

Reply:
Also sorted :)